### PR TITLE
Use a location port for service urls

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/client/test-run/index.js.mustache [hunk 1]
```diff
@@ -2,14 +2,16 @@
     if (window !== window.top)
         return;
 
+    var origin = location.protocol + '//' + location.host;
```
Comment (miherlosev) [inline src/client/test-run/index.js.mustache]: Just use the `location.origin` property.

File: src/browser/connection/index.js [hunk 2]
```diff
@@ -53,6 +53,10 @@ export default class BrowserConnection extends EventEmitter {
         this.statusDoneUrl = `${gateway.domain}/browser/status-done/${this.id}`;
         this.initScriptUrl = `${gateway.domain}/browser/init-script/${this.id}`;
 
+        this.heartbeatRelativeUrl  = `/browser/heartbeat/${this.id}`;
```
Comment (AlexanderMoskovkin) [inline src/browser/connection/index.js]:
```js
this.heartbeatUrl  = `${gateway.domain}${this.heartbeatRelativeUrl}`;
...
```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/functional/fixtures/regression/gh-2308/testcafe-fixtures/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+test('Load a top window on the cross-domain port', async t => {
+    await t
+        .switchToIframe('iframe')
+        .click('[type="submit"]')
+        .switchToMainWindow()
+        .click('body');
+});
+
+test('Second test', async () => {
+});
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/regression/gh-2308/testcafe-fixtures/index.js]: Let's add a comment and describe why do we need the second test

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
****

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**